### PR TITLE
Update train_HOG.cpp

### PR DESCRIPTION
**Pull request by sturkmen72**
### This pullrequest changes

Updates train_HOG.cpp

1.adds ability of testing trained detector with webcam
2. adds ability of using positive images when they have not exact same size ( when width and height are specified )
3.adds a new cmd line parameter "-f" indicates if the program will generate and use mirrored samples or not

### CODE REVIEW

**Comment by terfendail** [comment]
`computeHOG` use cropping to reduce input images to the size acceptable by `hog.compute`. Location of the region to crop is calculated as absolute value that fit to the center of the first image. If all images are of the same size this region fits all other images as well, but in case image sizes differ estimated region could crop different parts for different images. IMHO such behavior wouldn't be obvious for the user. I think it make sense to evaluate crop region for each image separately and clearly state in the help message the part of the image that will be processed in case detector size differs from the image size.
Also IMHO it make sense to check that all images are greater or at least equal to the size of the detector.

**Comment by sturkmen72** [comment]
@terfendail thank you for your comment. let me try to improve the PR according to your comment when i find a free time.

**Comment by sturkmen72** [comment]
@terfendail i tried to improve the code according to your comment with the commit : update1
if the code looks OK i will add some explanation about usage.

File: samples/cpp/train_HOG.cpp [hunk 1]
```diff
@@ -345,7 +349,7 @@ int main( int argc, char** argv )
         clog << "...[done]" << endl;
 
         labels.clear();
-        labels.assign( pos_lst.size(), +1 );
+        labels.assign( positive_count, +1 );
         labels.insert( labels.end(), neg_lst.size(), -1);
```
Comment (terfendail) [inline samples/cpp/train_HOG.cpp]: If there are small images in the negative set they will be discarded during `computeHOG` so the training set will be shorter than `neg_lst.size()`. Could you please use evaluation approach similar to the one in the first training step

Comment (terfendail) [reply samples/cpp/train_HOG.cpp]: Other parts of the code looks fine for me. It would be great to describe details of improved sample behavior

**Comment by terfendail** [comment]
Could you please squash commits into one

**Comment by sturkmen72** [comment]
@terfendail could you review updated code and merge if it is OK.
I am postponing adding the description for now. i have still some ideas to improve training process. i want to make some trials later and describe details of improved sample behavior

**Comment by terfendail** [comment]
👍

**Pull request merged**